constantine/eth_verkle_ipa/transcript_gen.nim

```diff
@@ -27,37 +29,72 @@ func newTranscriptGen*(res: var CryptoHash, label: openArray[byte]) =
   res.update(label)
 
 func messageAppend*(res: var CryptoHash, message: openArray[byte], label: openArray[byte]) =
-  res.init()
   res.update(label)
   res.update(message)
 
 func messageAppend_u64*(res: var CryptoHash, label: openArray[byte], num_value: uint64) = 
-  res.init()
   res.update(label)
   res.update(num_value.toBytes(bigEndian))
 
 func domainSeparator*(res: var CryptoHash, label: openArray[byte]) =
-  var state {.noInit.} : CryptoHash
-  state.update(label)
+  res.update(label)
 
 func pointAppend*(res: var CryptoHash, label: openArray[byte], point: EC_P) =
   var bytes {.noInit.}: array[32, byte]
-  if bytes.serialize(point) == cttCodecEcc_Success:
-    res.messageAppend(bytes, label)
+  let stat = bytes.serialize(point) 
+  doAssert stat == cttCodecEcc_Success, "Serialization Failure!"
+  res.messageAppend(bytes, label)
 
 func scalarAppend*(res: var CryptoHash, label: openArray[byte], scalar: matchingOrderBigInt(Banderwagon)) =
   var bytes {.noInit.}: array[32, byte]
 
-  if bytes.serialize_scalar(scalar) == cttCodecScalar_Success:
-    res.messageAppend(bytes, label)
+  let stat = bytes.serialize_scalar(scalar, littleEndian)
+  doAssert stat == cttCodecScalar_Success, "Issues with marshalling!"
+  res.messageAppend(bytes, label)
+
+func fromVerkleDigest(dst: var Fr[Banderwagon], src: array[32, byte]) : bool =
+  ## The input src byte array that we get here is 32 bytes
+  ## Which can be safely stored in a 256 BigInt
+  ## Now incase of the scalar overflowing the last 3-bits
+  ## it is converted from its natural representation
+  ## to the Montgomery residue form
+  ##
+  ## `mres` is overwritten. It's bitlength must be properly set before calling this procedure.
+  ##
+  ## Caller must take care of properly switching between
+  ## the natural and montgomery domain.
+  ## 
+  ## This is a function that can be called for added safety instead using the
+  ## `fromBig()` sequence of function calls in case of Banderwagon scalars
```

Comment:
All of this doesn't need `##`, it's internal details and so just `#` is sufficient so that they don't appear in user-facing doc and confuse them.

```
# Digests are interpreted as little-endian 256-bit integers.
# getMont converts them directly to Montgomery representation and reduces them modulo the curve order